### PR TITLE
Add patient search to frontend

### PR DESCRIPTION
**Pull request by nickscialli-usds**
## Related Issue or Background Info

- #1022 

## Changes Proposed

- Uses a debounced query to provide strings to existing graphql parameter
- Adds tests!

## Additional Information

- Added the ability to _not_ display the "Loading" indicator in `QueryWrapper`. The issue was that returning `<p>Loading...</p>` from `QueryWrapper` un-rendered its children from the DOM, blowing up all the local state those children had (e.g., query text!).
- During testing, I noticed that we connected to redux (via `useSelector`) in `ManagePatientsContainer`, `ManagePatients`, and `PatientUpload`. This seemed to defeat the purpose of having a centralized store, so I removed the redux connections from `ManagePatients` and `PatientUpload` and just connected at the container level.

### CODE REVIEW

**Comment by nickscialli-usds** [comment]
Need to troubleshoot why this isn't working on the test env

**Comment by nickscialli-usds** [comment]
OK, fixed up routing in the test environment. @katiealoisi and @hmyers-cms this is available for review on https://test.simplereport.gov

**Comment by hmyers-cms** [comment]
I think this looks great! My only comment is that I would love for the drawer to animate open/closed if that isn't too difficult,  but we shouldn't block on it

File: frontend/src/app/patients/ManagePatients.test.tsx [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    expect(await screen.findByText(patients[0].lastName, { exact: false }));
+    const btn = await screen.findByText("Filter", { exact: false });
+    fireEvent.click(btn);
+    const input = await screen.findByLabelText("Person");
+    fireEvent.change(input, { target: { value: "Al" } });
+    await waitForElementToBeRemoved(() =>
+      screen.queryByText("Abramcik", { exact: false })
+    );
+    expect(await screen.findByText(patients[1].lastName, { exact: false }));
+    expect(await screen.findByText(patients[2].lastName, { exact: false }));
```
Comment (ClydeDroid) [inline frontend/src/app/patients/ManagePatients.test.tsx:60]: Thanks for adding this! Glad we are starting to flesh out unit tests on the frontend!

**Review by ClydeDroid — APPROVED** [review]
Looks great! Checked it out in test as well, works like a charm!

**Comment by katiealoisi** [comment]
For some reason, I am only seeing the Filter when I select facility b (don't see it when selecting facility a). With that said, Heather can see it for both, so it could just be something on my end.

I'm also running into some issues when searching by full name - take Emmy O Alabaster for example. I see her record appear as soon as I type "Emm" but there are no results found if I type "Emmy O" or "Emmy O Alabaster" or "Emmy Alabaster".

**Comment by nickscialli-usds** [comment]
>For some reason, I am only seeing the Filter when I select facility b (don't see it when selecting facility a). With that said, Heather can see it for both, so it could just be something on my end.

Probably some kind of caching behavior

>I'm also running into some issues when searching by full name - take Emmy O Alabaster for example. I see her record appear as soon as I type "Emm" but there are no results found if I type "Emmy O" or "Emmy O Alabaster" or "Emmy Alabaster".

I don't think we currently have the backend wired up to do anything but search the first name and the last name independently. @TomNUSDS could speak more to it. I'm sure we could do full name searches but would just require some work (and preferably a new feature requirement to not hold this one up).

**Comment by TomNUSDS** [comment]
@nickscialli-usds @katiealoisi Multi-word searching is not yet supported. See https://github.com/CDCgov/prime-simplereport/issues/1130

**Pull request merged**